writer.go

```diff
@@ -18,6 +18,12 @@ import (
 	"github.com/segmentio/encoding/thrift"
 )
 
+const (
+	// The uncompressed page size is stored as int32 and must not be larger than the
+	// maximum int32 value (see format.PageHeader).
+	maxUncompressedPageSize = int(^uint32(0) >> 1)
```

Comment:
Any concerns about using a named constant?
```suggestion
	maxUncompressedPageSize = math.MaxInt32
```

Reply:
Not at all, I completely missed that those constants were added to the `math` package (TIL)